source/dealii/dealii_hierarchy_helpers.cc

```diff
@@ -159,11 +157,20 @@ DealIIHierarchyHelpers<dim, VectorType>::build_restrictor(
 
             // Put the result in the matrix
             // Compute the map between the local and the global dof indices.
-            std::vector<dealii::types::global_dof_index> dof_indices_map =
-                amge.compute_dof_index_map(patch_to_global_map,
-                                           agglomerate_dof_handler);
+            local_copy_data.rows.resize(n_local_eigenvectors);
+
+            local_copy_data.cols = amge.compute_dof_index_map(
+                patch_to_global_map, agglomerate_dof_handler);
+            auto const &dof_indices_map = local_copy_data.cols;
             unsigned int const n_elem = dof_indices_map.size();
+
+            local_copy_data.values_per_row.resize(n_local_eigenvectors);
+            std::fill(local_copy_data.values_per_row.begin(),
+                      local_copy_data.values_per_row.end(),
+                      std::vector<dealii::TrilinosScalar>(n_elem));
```

Comment:
Could this 4 lines replaces by:
```c++
local_copy_data.values_per_row.resize(n_local_eigenvectors, std::vector<dealii::TrilinosScalar>(n_elem));
```
or do you need a clean reset?

Reply:
I need a clean reset. Otherwise, we would accumulate values across patches corresponding to different degrees of freedom.

Reply:
Please add a comment to capture this.